recipes/graphene/all/conanfile.py

```diff
@@ -73,6 +74,11 @@ def build_requirements(self):
         self.tool_requires("meson/1.1.0")
         if not self.conf.get("tools.gnu:pkg_config", default=False):
             self.tool_requires("pkgconf/1.9.3")
+        # if self.options.with_glib and cross_building(self):
+        if self.options.with_glib and not can_run(self):
+            # Same Tool as for requirements to find the glib compiler
+            # This is required for dual-profile and cross builds.
```

Comment:
I don't think the `not can_run` check is necessary is it? As a `tool_requires`, `glib` will be built for the build profile and not the host profile, therefore it should be possible to run it on the build machine. Am I missing something?

```suggestion
        if self.options.with_glib:
```

Reply:
As I've mentioned above, I think conditional tool_requires on can_run() or similar are not necessary.  If this recipe makes use of the executables inside the glib package at _build_ time - that is, those scripts or executables are used exclusively during a build process, my suggestion still stands as follows:

```
def requirements(self):
    self.options.with_glib:
        self.requires("glib/2.76.2")

def build_requirements(self):
    self.options.with_glib:
        self.tool_requires("glib/2.76.2")
```

if there are known version compatibility issues between the "executables" in the glib package and the library (such as those that exist with protobuf), we have added the `<host_version>` version alias to ensure the versions are aligned - More details here: https://docs.conan.io/2/reference/conanfile/methods/build_requirements.html?highlight=host_version#host-version - this was also backported to Conan 1.x and tested with 1 profile.

```
def requirements(self):
    self.options.with_glib:
        self.requires("glib/2.76.2")

def build_requirements(self):
    self.options.with_glib:
        self.tool_requires("glib/<host_version>")
```

Note that when dealing with 2 profiles, this may still bring 2 different package IDs for host and build contexts if the profiles are different. This is _fine_ and intended, if the user wants to ensure the same when _not_ crossbuilding, they can tell Conan to use the same profile in both scenarios and pass the same options to both host and build.

For the case of `glib` in particular, some of these executables are actually python scripts - they are going to be the same regardless of package ID. 

Given that there are no known version compatibility issues and that `<host_version>` should take care of keeping versions aligned in the presence of version overrides elsewhere - I see no reason to have any more complex logic, unless specific reproducible cases can be provided.